src/hotspot/share/oops/stackChunkOop.cpp

```diff
@@ -441,32 +441,33 @@ void stackChunkOopDesc::fix_thawed_frame(const frame& f, const RegisterMapT* map
 template void stackChunkOopDesc::fix_thawed_frame(const frame& f, const RegisterMap* map);
 template void stackChunkOopDesc::fix_thawed_frame(const frame& f, const SmallRegisterMap* map);
 
-void stackChunkOopDesc::copy_lockstack(oop* dst) {
-  int cnt = lockstack_size();
-
-  if (!(is_gc_mode() || requires_barriers())) {
-    oop* lockstack_start = (oop*)start_address();
-    for (int i = 0; i < cnt; i++) {
-      dst[i] = lockstack_start[i];
-      assert(oopDesc::is_oop(dst[i]), "not an oop");
+void stackChunkOopDesc::transfer_lockstack(oop* dst) {
+  const bool requires_gc_barriers = is_gc_mode() || requires_barriers();
```

Comment:
Given how careful we are in `Thaw` to not call `requires_barriers()` twice and use `_barriers` instead it would probably be nicer to pass in `_barriers` as a bool. 

There is only one other place we do the extra call and it is in `fix_thawed_frame`, but that only happens after we are committed to the slow path, so it might be nice for completeness, but should be negligible for performance. Here however we might still be in our new "medium" path where we could still do a fast thaw.

Reply:
Good, passed as argument now.